### PR TITLE
fix: account currency hook return value

### PR DESCRIPTION
**Pull request by frosso**
Fixes #

#### Changes proposed in this Pull Request

The return value of the hook is a string, which is erroneously deconstructed - so only the first character is being returned (instead of the whole string).
So instead of `USD`, only `U` is returned.

#### Testing instructions

*

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/payment-methods/index.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ const PaymentMethods = () => {
 
 	const [ , updateSelectedPaymentMethod ] = useSelectedPaymentMethod();
 
-	const [ stripeAccountDomesticCurrency ] = useAccountDomesticCurrency();
```
Comment (frosso) [inline client/payment-methods/index.js:95]: Before: the returned value of `useAccountDomesticCurrency()` ( a string ), was being deconstructed, and only the first character was being returned.

File: client/payment-methods/index.js [hunk 2]
```diff
@@ -92,7 +92,7 @@ const PaymentMethods = () => {
...
+	const stripeAccountDomesticCurrency = useAccountDomesticCurrency();
```
Comment (frosso) [inline client/payment-methods/index.js:95]: After: the whole string is returned.

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7872` or branch name `fix/account-currency-hook` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/account-currency-hook) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 6b9a09d56d7282f4b1d767ad673230d4b3ef26d6
- Build time: 2023-12-11 21:36:28 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by eduardoumpierre — APPROVED** [review]
It works as expected and the tests are passing. The change makes sense as the `useAccountDomesticCurrency` hook is being used in the same way it was already being used on:

https://github.com/Automattic/woocommerce-payments/blob/6b9a09d56d7282f4b1d767ad673230d4b3ef26d6/client/components/payment-methods-checkboxes/payment-method-checkbox.tsx#L37

Thanks for fixing it. LGTM! :shipit:

**Pull request merged**